### PR TITLE
Add docker compose file to the root of the project

### LINKED ISSUE #564: [Feature Request]: Integrating docker compose into the project
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

I would like to add docker-compose.yml file to the root of the project.

### Use Case

You can set the desired amount of containers counts, their builds, and storage designs, and then with a single set of commands you can build, run and configure all the containers. Docker Compose is great for development, testing, and staging environments, as well as continuous integration workflows.

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by haseebzaki-07**
# Pull Request

### Title
This pr adds docker-compose.yaml at the root of the project

### Description
fixes #564 

### Checklist 

- [x ] I have tested the changes locally
- [x ] Documentation has been updated (if necessary)
- [x] Changes are backward-compatible

Footer

### CODE REVIEW

**Comment by VaibhavArora314** [comment]
Where are the dockerfiles? I think it will not work without dockerfiles if I am not wrong.

**Comment by haseebzaki-07** [comment]
I have created a separate issue for the Dockerfile(s) issue #690. I can write the Dockerfiles(s) there.

**Comment by haseebzaki-07** [comment]
I have added the Dockerfiles

**Review by VaibhavArora314 — COMMENTED** [review]
Looks good to me, just one change. Instead of passing the .env file to the backend container, only pass the database url of the mongodb container. As for the rest of the environment variables, I will add default values.

**Comment by haseebzaki-07** [comment]
Sure I'll do that

**Comment by haseebzaki-07** [comment]
I have updated the docker compose having only the DATABASE_URL as the env-variable

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+version: '3.8'
```
Comment (akbatra567) [inline docker-compose.yml:1]:
```suggestion
```
Remove version field as in latest update docker has made it obsolete, removing the line fixes the issue, link for reference: https://forums.docker.com/t/docker-compose-yml-version-is-obsolete/141313

**Review by akbatra567 — CHANGES_REQUESTED** [review]

**Review by VaibhavArora314 — APPROVED** [review]

**Review by akbatra567 — APPROVED** [review]

**Pull request merged**